packages/loader/container-loader/src/container.ts

```diff
@@ -2436,11 +2453,92 @@ export class Container
 	}
 }
 
+/**
+ * Diagnostics information about a given connection instance (single attempt/clientId),
+ * traversing the four connection states (disconnected, establishingConnection, catchingUp, connected).
+ *
+ * Includes what the current connection state is, and what is happening within that state
+ *
+ * @beta
+ */
+export interface ConnectionDiagnostics {
+	/**
+	 * The client id assigned for this connection by the ordering service
+	 * Only set when currentState is "catchingUp" or "connected"
+	 */
+	clientId?: string;
```

Comment:
Why would we want to expose it? It feels that until we are fully connected, it should not be exposed.
Also what is the scope here? I.e. if we keep losing connection and establish new one, but never reach connected state - should clientID keep changing as we cycle through connections?

Reply:
Could be useful for diagnostics with service, if we get to catchingUp state (that's where this will be set) but keep getting an error then, and never actually apply the clientId to the Container.